test/models/thesis_test.rb

```diff
@@ -765,4 +765,31 @@ class ThesisTest < ActiveSupport::TestCase
     new_thesis.update(title: 'updated')
     refute new_thesis.new_thesis?
   end
+
+  test 'supports one DSpace metadata attachment per thesis' do
+    thesis = theses(:one)
+
+    # Add metadata required for DspaceMetadata class
+    degree = thesis.degrees.first
+    degree.degree_type_id = degree_types(:bachelor).id
+    degree.save
```

Comment:
I'm not sure why we need this part? Does the fixture for the Degree that this Thesis uses not using a valid Degree Type? And if so, does fixing that break a lot of other tests?

Reply:
Good point. All tests pass if we associate the bachelor degree type with the first degree fixture, so I think it's fine (and preferable) to do it that way. I pushed up a commit updating that here and in the DspaceMetadata tests.